ding/model/template/diffusion.py

```diff
@@ -0,0 +1,617 @@
+from typing import Union, List, Dict
```

Comment:
I think you can add some unit tests to make the code easier to understand.
And it's convenient to check the rightness for each module.